### PR TITLE
Match pact js lint

### PR DESCRIPTION
**Pull request by TimothyJones**
This is like https://github.com/pact-foundation/pact-js-core/pull/425 and
https://github.com/pact-foundation/pact-js/pull/1027 - it simply tightens the lint settings to match pact-js and pact-js core.

Key points:

* This is a Jest project, so it uses jest settings rather than mocha.
    * If you use this as a base for other jest projects, make sure you enable `jest/expect-expect` - it is only disabled because we have tests that always throw to test "this shouldn't run"
* Also I tightened the tsconfig settings to match pact-js-core
* Some code is now duplicated because you can't import from tests (this results in the tests being run twice). An improvement would be to have a `__fixtures__` or something that the build skips.

### CODE REVIEW

**Comment by TimothyJones** [comment]
Doing the same treatment to mocha pact is left as an exercise for the reader

**Comment by YOU54F** [comment]
Thanks @TimothyJones, this looks good, would you be able to resolve the conflicts and we can get this merged :)

File: package.json [hunk 1]
```diff
@@ -41,10 +42,20 @@
     "@commitlint/config-conventional": "^10.0.0",
     "@pact-foundation/pact": "^10.0.2",
     "@pact-foundation/pact-js-prettier-config": "^1.0.0",
+    "@tsconfig/node14": "^1.0.3",
```
Comment (YOU54F) [inline package.json:45]: separate to this PR, but probably worth doing an update to drop 14.x and update the node engines field

**Review by YOU54F — APPROVED** [review]
nice one, sorry this got left to the side :)

File: package.json (hunk 1 above)
Comment (TimothyJones) [reply package.json:45]: yep

**Pull request merged**